runtime/src/main/java/org/corfudb/runtime/object/transactions/SnapshotTransactionalContext.java

```diff
@@ -91,7 +92,11 @@ public void addTransaction(AbstractTransactionalContext tc) {
 
     @Override
     public long obtainSnapshotTimestamp() {
-        return getBuilder().getSnapshot();
+        if (getBuilder().getSnapshot() == Address.NON_ADDRESS) {
```

Comment:
Is there a new possibility introduced where SnapshotTransactionContext's obtainSnapshotTimestamp() is called and the getBuilder().getSanpshot() is a non address?

Reply:
That would seem to be a bug in application logic (the application provided an illegal address in the builder). Perhaps we could check for that in the constructor in the future, but I don't see any behavior change.

Reply:
I tried something like : 

+    public TransactionBuilder setSnapshot(long snapshot) {
+        if (snapshot <= Address.NON_ADDRESS) {
+            throw new IllegalArgumentException("Invalid snapshot: the timestamp has to be greater than -1");
+        }
+
+        this.snapshot = snapshot;
+        return this;
+    }

but some tests were failing, because -1 is actually a valid snapshot (i.e. empty). We can't force > -1 because [0, inf+) might have not been written.